### ISSUE TITLE
AuthUnknown error when running code in Windows Service

### ISSUE THREAD

**Issue by umar-ulabs** [flamencist/ldap4net#112]
**The connection fails when using public IP or domain of the AD server**
 Everything works fine except when code is running inside a windows service and the server address is public one(port forwarding to 389 or 636).
The error return is AuthUnknown which is result code 86
Following are the connection settings:
` var connection = new LdapConnection();
                var isSecure = _config.UseSSL || _config.Port == 636 || _config.Port == 3269;
                connection.Connect(_config.Server, _config.Port, isSecure ?
                    LdapSchema.LDAPS : LdapSchema.LDAP);
                if (_configUseSSL)
                {
                    newConnection.TrustAllCertificates();
                }

                var cred = new LdapCredential();
                cred.UserName = _config.Username;
                cred.Password = _config.Password;
                await connection.BindAsync(LdapAuthType.Negotiate , cred);`
 Thanks

**Comment by umar-ulabs** [comment]
Right. What I did is try catch for couple of auth types. If one fails ,use different auth type. It seems working well that way. The OR on auth types didnt work like Auth Type Negotiate | Auth Type Simple, but using try catch to try different auth types seems working for now

**Comment by flamencist** [comment]
Yes. Its lightweight library without most features like cache or reliability
